webapp/src/js/controllers/reports.js

```diff
@@ -49,6 +51,34 @@ angular
 
     var lineage = lineageFactory();
 
+    // Render the facilities hierarchy as the user is scrolling through the list
+    // Initially, don't load/render any
+    $scope.totalFacilitiesDisplayed = 0;
+    $scope.facilities = [];
+
+    // Load the facilities hierarchy and render one district hospital 
+    // when the user clicks on the filter dropdown
+    $scope.monitorFacilityDropdown = () => {
+      PlaceHierarchy()
+        .then(function(hierarchy) {
+          $scope.facilities = hierarchy;
+          $scope.totalFacilitiesDisplayed += 1;
+        })
+        .catch(function(err) {
+          $log.error('Error loading facilities', err);
+        });
+
+      $document[0].querySelector('#facilityDropdown span.dropdown-menu > ul')
+                  .addEventListener('scroll', (event) => {
+        // visible height + pixel scrolled >= total height - 100
+        if (event.target.offsetHeight + event.target.scrollTop >= event.target.scrollHeight - 100) {
+          $scope.$apply(function() {
+            $scope.totalFacilitiesDisplayed += 1;
```

Comment:
Why add 1? I feel like it'd be slightly more intuitive to add pages of 10-20?

Reply:
These are for the parent facilities ... there are a total of 25 parent facilities (district hospital) and 5079 child facilities (health centers). The scrolling is less smooth as the number of parent/child facilities to render is huge.

Reply:
Ahh I see, so each 1 added here renders approximately 1/25th of all the facilities. That works for me!